build-tools/scripts/ImportExportDocs.targets

```diff
@@ -0,0 +1,96 @@
+        DestinationFiles="$(_XamarinDocsPath)docs\android\deploy-test\building-apps\build-process.md"
+        SkipUnchangedFiles="true"
+    />
+  </Target>
+  <Target Name="ImportXamarinEngineeringDocs">
+    <ItemGroup>
+      <_MessageDocumentationFile
+          Include="$(_XamarinEngineeringDocsPath)docs\android\errors-and-warnings\*.md"
+          Exclude="$(_XamarinEngineeringDocsPath)docs\android\errors-and-warnings\TOC.md;$(XamarinEngineeringDocsPath)docs\android\errors-and-warnings\index.md"
+      />
+      <_MessageDocumentationImages
+          Include="$(_XamarinEngineeringDocsPath)docs\android\errors-and-warnings\images\*"
+      />
+    </ItemGroup>
+    <Copy
+        SourceFiles="@(_MessageDocumentationFile)"
+        DestinationFolder="$(_TopDirImportExportDocs)Documentation\guides\messages"
+        SkipUnchangedFiles="true"
+    />
+    <Copy
+        SourceFiles="@(_MessageDocumentationImages)"
+        DestinationFolder="$(_TopDirImportExportDocs)Documentation\guides\messages\images"
+        SkipUnchangedFiles="true"
+    />
+    <Copy
+        SourceFiles="$(_XamarinEngineeringDocsPath)docs\android\errors-and-warnings\index.md"
+        DestinationFiles="$(_TopDirImportExportDocs)Documentation\guides\messages\README.md"
+        SkipUnchangedFiles="true"
+    />
+  </Target>
+  <Target Name="ExportXamarinEngineeringDocs">
+    <ItemGroup>
+      <_MessageDocumentationFile
+          Include="$(_TopDirImportExportDocs)Documentation\guides\messages\*.md"
+          Exclude="$(_TopDirImportExportDocs)Documentation\guides\messages\README.md"
+      />
+      <_MessageDocumentationImages
+          Include="$(_TopDirImportExportDocs)Documentation\guides\messages\images\*"
+      />
+    </ItemGroup>
+    <Copy
+        SourceFiles="@(_MessageDocumentationFile)"
+        DestinationFolder="$(_XamarinEngineeringDocsPath)docs\android\errors-and-warnings\"
+        SkipUnchangedFiles="true"
+    />
+    <Copy
+        SourceFiles="@(_MessageDocumentationImages)"
+        DestinationFolder="$(_XamarinEngineeringDocsPath)docs\android\errors-and-warnings\images"
+        SkipUnchangedFiles="true"
+    />
+    <Copy
+        SourceFiles="$(_TopDirImportExportDocs)Documentation\guides\messages\README.md"
+        DestinationFiles="$(_XamarinEngineeringDocsPath)docs\android\errors-and-warnings\index.md"
+        SkipUnchangedFiles="true"
+    />
+    <ReadLinesFromFile File="$(_TopDirImportExportDocs)Documentation\guides\messages\README.md">
+      <Output
+          TaskParameter="Lines"
+          ItemName="_MessageDocsReadmeLines"
+      />
+    </ReadLinesFromFile>
+    <ItemGroup>
+      <_MessageDocsTocLines Include="# [Errors and Warnings](index.md)" />
+      <_MessageDocsTocLines
+          Include="$([System.Text.RegularExpressions.Regex]::Replace($([System.Text.RegularExpressions.Regex]::Replace('%(_MessageDocsReadmeLines.Identity)', '^\s*\+\s', '### ')), '\):.*', ')'))"
```

Comment:
Generating the `TOC.md` file from the `README.md` file via regular expression replacements might be more fragile than its worth. Another option would be to remove this generation step and just put a manually maintained `TOC.md` file directly in the `Documentation/messages` directory alongside the other files. I'd be A-OK with that, if it seems like the better approach.